content/guides/pedestal-with-component.adoc

```diff
@@ -1,16 +1,419 @@
 
-Pull requests welcomed!
+== What You Will Learn
+After reading this guide you will be able to:
 
+- Create a Component-based service using Pedestal.
+- Test your service using Pedestal's test helpers.
 
-__TODO__ Incorporate information from https://github.com/pedestal/pedestal/issues/366
+== Guide Assumptions
+This guide is for users who are familiar with Clojure, Pedestal,
+Component and Boot. If you are new to Pedestal, you may want to go back to the
+link:hello-world[Hello World guide]. If you're new to Component, you
+should definitely https://github.com/stuartsierra/component[check it
+out] first.
+
+== Getting Help if You're Stuck
+If you get stuck at any point in this guide, please submit an
+https://github.com/pedestal/docs/issues[issue] about this guide or hop
+over to the
+https://groups.google.com/forum/#!forum/pedestal-users[mailing list]
+and raise your hand there.
+
+This guide shows fragments of code as we add them. Sometimes it helps
+to see the whole thing at once, so you can always check out
+<<_the_whole_shebang,the whole shebang>> at the end of this guide.
+
+== Where We Are Going
+In this guide, we're going to step through creating a Pedestal service
+using Component. We'll start off by creating a Pedestal component and
+wire it into a Component system map. We'll then proceed to testing our
+service.
+
+== Before We Begin
+What responsibilities do we want the Pedestal component to have? For
+the purposes of this guide, it should only be responsible for the
+initialization of the Pedestal Provider and starting of the
+server. It's configuration, the Pedestal service map, should be a
+component dependency and should not be managed by the Pedestal
+component.
+
+The Pedestal component should also be amiable to testing through
+Pedestal's testing facilities. This should only require a Pedestal
+service-fn to be available.
+
+== A Simple Pedestal Component
+
+Let's start with the Pedestal component. Create a `src` directory with
+a `pedestal.clj` file.
+
+[source, clojure]
+.src/pedestal.clj
+
+----
+include::component/src/pedestal.clj[tags=ns]
+----
+<1> Create a _pedestal_ namespace to house the Pedestal component.
+<2> We need to require `com.stuartsierra.component` to implement the
+`start` and `stop` `Lifecycle` methods.
+<3> We need to require the `io.pedestal.http` for server creation,
+starting and stopping.
+
+Let's start implementing the component.
+----
+include::component/src/pedestal.clj[tags=component-init]
+----
+<1> Create a _Pedestal_ record. This record will contain a number of
+ fields, the first field is _service-map_ which will hold the Pedestal
+ service map component dependency.
+<2> The next field is _start-fn_, this will be passed in through the
+ component constructor and will be used to start the Pedestal
+ server. This function must accept one parameter, the Pedestal service
+ map, and return a Pedestal service map.
+<3> The next field is _stop-fn_, like _start-fn_, the stop function
```

Comment:
It might also help to explain whether we are going to write the start-fn and stop-fn or whether we're just going to pass something in here. At this point, the reader doesn't know how to connect the dots.